### ISSUE TITLE
No sound after playing recording (MacCatalyst)

### ISSUE THREAD

**Issue by MaikelW** [jfversluis/Plugin.Maui.Audio#126]
I'm using Blazor Hybrid with Razor pages and have some issues with playing back recorded audio on Mac. It looks like recording works fine, since the audio control loads the audio stream successfully and displays its length in seconds. However, when playing back the audio, no sound is being produced. What could the issue be? 

**AudioService.cs**
```
public class AudioService : IAudioService
{
    private IAudioManager _audioManager;
    private IAudioRecorder _audioRecorder;
    public bool IsRecording => _audioRecorder.IsRecording;
    public AudioService(IAudioManager audioManager)
    {
        _audioManager = audioManager;
        _audioRecorder = audioManager.CreateRecorder();
    }

    public async Task StartRecordingAsync()
    {
        await _audioRecorder.StartAsync();
    }

    public async Task<Stream> StopRecordingAsync()
    {
        var recordedAudio = await _audioRecorder.StopAsync();

        // Tried code below as well, but also didn't work
        // var player = _audioManager.CreatePlayer(recordedAudio.GetAudioStream());
        // player.Play();

        return recordedAudio.GetAudioStream();
    }
}
```

**Home.razor**
```
@page "/"
@inject AudioService AudioService;

<h1>Hello, world!</h1>

Welcome to your new app.

<div class="flex-column mt3">
    <div class="f5 mb2">
        Record an audio
    </div>
    <div class="f5 mb2">
        @if (!AudioService.IsRecording)
        {
            <button @onclick="StartRecording"> Start</button>
        }
        else
        {
            <button @onclick="StopRecording"> Stop</button>
        }
    </div>
    <audio controls src="@AudioSource" type="audio/mpeg">
        Your browser does not support the audio element.
    </audio>


</div>

@code {
    private Stream? recordedAudioStream;
    private string? AudioSource { get; set; }
    public async void StartRecording()
    {
        if (await Permissions.RequestAsync<Permissions.Microphone>() != PermissionStatus.Granted)
        {
            // Inform user to request permission
        }
        else
        {
            await AudioService.StartRecordingAsync();

        }

    }
    public async void StopRecording()
    {
        recordedAudioStream = await AudioService.StopRecordingAsync();


        var audioBytes = new byte[recordedAudioStream.Length];
        
        await recordedAudioStream.ReadAsync(audioBytes, 0, (int)recordedAudioStream.Length);
        var base64String = Convert.ToBase64String(audioBytes);

        AudioSource = $"data:audio/mpeg;base64,{base64String}";

        StateHasChanged();
    }
}
```

**Comment by bijington** [comment]
What version are you using?

I would expect the sounds to play but there are some things that you can check:

- is the system sound muted?
- can you try the Configure playback options section from this page https://github.com/jfversluis/Plugin.Maui.Audio/blob/main/docs/audio-player.md

**Comment by MaikelW** [comment]
I'm using version 3.0.0. Double checked if my system was muted, which wasn't the case. Also implemented those playback options, but still no sound unfortunately...

**Comment by bijington** [comment]
I am not sure here, I can confirm that using the CreatePlayer method and playing the recording through that player works as expected.

**Comment by MaikelW** [comment]
Weird... Could you send me your example to see if I can get that to work? And are you using Razor pages?

**Comment by bijington** [comment]
> Weird... Could you send me your example to see if I can get that to work? And are you using Razor pages?

It's just using the sample app in this repository. It's not using Razor pages, just plain MAUI but the playback code is agnostic to the UI framework so it should work in Razor. It just won't render the playback controls for you

**Comment by MaikelW** [comment]
I'll get the sample app up and running and will then come back to you. Thanks for your help so far!

**Comment by MaikelW** [comment]
Seems to be working just fine in the sample app indeed. I'll see if I can figure out why it's not working in my razor pages app.

**Comment by bijington** [comment]
Good luck and let us know how you get on

**Comment by MaikelW** [comment]
Alright, I think it might have something to do with the thread that is responsible for playing the audio. When I record and play back the audio multiple times in a row, sometimes it works. Might be interesting to see if someone else also experiences this behavior when using razor pages.

**Comment by Khongchai** [comment]
I'm using razor pages and am also facing this issue. Did you manage to get it working? @MaikelW 

I checked the generated file from the audio recorder, the entire file basically just contains silence (a bunch of 0s in the hex data except the header), but with correct duration, it's as if the microphone was not used somehow but the bytes are being streamed to the file. The yellow microphone icon at the top right 

<img width="78" alt="image" src="https://github.com/user-attachments/assets/2673e0a8-716b-4fd0-b7f4-83ba9728a953"> _<< expected to see this icon but did not_


didn't even show up during recording.


And like @MaikelW , I also tried the official sample project and it works fine.

**Comment by MaikelW** [comment]
Nope, unfortunately I wasn't able to fix the issue in my razor pages project. But I'm indeed experiencing the exact same thing as you are @Khongchai. Sometimes the audio file is indeed filled with just 0s, but with the correct duration.

**Comment by ronnycsharp** [comment]
same problem. how to solve it?

**Comment by ronnycsharp** [comment]
It's fixed now. I've searched hours.
To solve it, put this into your entitlements.plist

        <key>com.apple.security.device.audio-input</key>
        <true/>

**Comment by jfversluis** [comment]
Need to make a note of this in the README
